src/driver/driver.cpp

```diff
@@ -39,4 +62,42 @@ void Driver::dumpRegFile(std::ostream &outs) const {
   }
 }
 
+std::vector<Instruction> Driver::lookupBB(paddress_t addr) {
+  if (!m_nativeBBChache.count(addr)) {
+    Instruction instr;
+    paddress_t cur_addr = addr;
+    std::vector<Instruction> res;
+    do {
+      word_t cur_word = m_hart.memory.readWord(cur_addr);
+      instr = m_decoder.decode(cur_word);
+      res.push_back(instr);
+      cur_addr += 4;
+    } while (!is_terminate(instr.id));
+
+    m_nativeBBChache.emplace(addr, res);
+  }
+  return m_nativeBBChache.at(addr);
+}
+
+bool Driver::is_terminate(InstrId id) {
+  switch (id) {
+  case InstrId::BEQ:
+  case InstrId::BGE:
+  case InstrId::BGEU:
+  case InstrId::BLT:
+  case InstrId::BLTU:
+  case InstrId::BNE:
+  case InstrId::EBREAK:
+  case InstrId::ECALL:
+  case InstrId::JAL:
+  case InstrId::JALR:
+  case InstrId::PAUSE:
+  case InstrId::SBREAK:
+  case InstrId::SCALL:
+    return true;
+    break;
+  }
+  return false;
+}
```

Comment:
I thinks we can avoid using such terminator checks. We make can make callback in Hart smth like `closeBB` that will tell hart, that this instruction is terminator and BB should be closed. And instead of using switch, each terminator instruction will call this function. For example
```C++
void exec_BEQ(Instruction &instr, Hart &hart) {
  ... // handler code
  hart.closeBB(); // Tell hart that this instruction is terminator
}
```
In that way we don't need to use any conditions for terminator checking. This is not good from codestyle point of view, but probably will boost simulator performance.
Actually, if Decoder will be able to identify terminators, than we can do some compile-time magic to insert such callbacks automatically, and then it will be nice both for codestyle and perf
What do you think @Dmitrryy @uslsteen